File: backend/src/component/Order.js
import React from "react";
import {
  List,
  Datagrid,
  TextField,
  Edit,
  SimpleForm,
  EditButton,
  TextInput,
  Create,
  DeleteButton,
  NumberInput,
} from "react-admin";

export const listOrder = (props) => (
  <List {...props}>
    <Datagrid>
    <TextField source="title" />
      <TextField source="user_id" />
      <TextField source="name" />
      <TextField source="email" />
      <TextField source="phone" />
      <TextField source="address" />
      <TextField source="note" />
      <TextField source="created_at" />
      <TextField source="updated_at" />
      <TextField source="created_by" />
      <TextField source="updated_by" />
      <TextField source="status" />
      <TextField source="description" />
      <TextField source="fullname" />
      <TextField source="title" />
      <TextField source="total_money" />
      <EditButton />
      <DeleteButton />
    </Datagrid>
  </List>
);

export const editOrder = (props) => {

  const getCurrentDateTime = () => {
    const now = new Date();
    return now.toISOString();
  };

  
  return (
  <Edit {...props}>
    <SimpleForm>
      <TextInput source="title" />
      <TextInput source="user_id" />
      <TextInput source="name" />
      <TextInput source="email" />
      <TextInput source="phone" />
      <TextInput source="address" />
      <TextInput source="note" />
      <TextInput source="created_at" defaultValue={getCurrentDateTime()} />
      <TextInput source="updated_at" defaultValue={getCurrentDateTime()} />
      <TextInput source="status" />
      <TextInput source="description" />
      <TextInput source="fullname" />
      <TextInput source="title" />
      <TextInput source="total_money" />
    </SimpleForm>
  </Edit>
)
  };

export const createOrder = (props) => {
  const getCurrentDateTime = () => {
    const now = new Date();
    return now.toISOString();
  };
  return (
  <Create {...props}>
    <SimpleForm>
    <TextInput source="title" />
      <TextInput source="user_id" />
      <TextInput source="name" />
      <TextInput source="email" />
      <TextInput source="phone" />
      <TextInput source="address" />
      <TextInput source="note" />
      <TextInput source="created_at" defaultValue={getCurrentDateTime()} />
      <TextInput source="updated_at" defaultValue={getCurrentDateTime()} />
      <NumberInput source="status" />
      <TextInput source="description" />
      <TextInput source="fullname" />
      <NumberInput source="total_money" />
    </SimpleForm>
  </Create>
)
  };

File: APIJavaSpringBoot/src/main/java/com/example/example3/entity/Order.java
package com.example.example3.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.Date;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "db_order")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private int user_id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false)
    private String address;

    @Column
    private String note;

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date created_at;

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date updated_at;
    
    @Column(nullable = false)
    private String created_by;

    @Column(nullable = false)
    private String updated_by;

    @Column(nullable = false)
    private String status;

    @Column
    private String description;

    @Column
    private String fullname;

  

    @Column
    private String title;

    @Column(nullable = false)
    private BigDecimal total_money;



    @JsonIgnore
 

    public void setTotalMoney(BigDecimal total_money) {
        this.total_money = total_money;
    }

    public BigDecimal getTotalMoney() {
        return total_money;
    }




}
